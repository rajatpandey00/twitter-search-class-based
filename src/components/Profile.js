import React from 'react'
import { Media, Spinner } from 'react-bootstrap'
import Tweets from './Tweets'
import '../styles/search.scss'

// @flow
export default ({ userDetails, tweets, tweetsFetched }) => {
  return (
    <div>
      <Media>
        <img
          width={64}
          height={64}
          className="mr-3"
          src={userDetails.profile_image_url}
          alt={userDetails.name}
        />
        <Media.Body>
          <h5>{userDetails.name}</h5>
          <p>
            {userDetails.description}
          </p>
          <h5>Tweets</h5>
          <div>
            {!tweetsFetched && <Spinner animation="border" variant="primary" />}
            {tweetsFetched && <Tweets className="profile-tweets" tweets={tweets} />}
          </div>
        </Media.Body>
      </Media>
    </div>
  )
}