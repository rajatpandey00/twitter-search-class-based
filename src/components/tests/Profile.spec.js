import React from 'react';
import renderer from 'react-test-renderer';
 
import Profile from '../Profile';
 
describe('Profile', () => {
    const mockUser = { profile_image_url : 'blabl blah', tweets:[{ text: 'test tweet'}] }
  test('snapshot renders', () => {
    const component = renderer.create(<Profile userDetails={mockUser}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});