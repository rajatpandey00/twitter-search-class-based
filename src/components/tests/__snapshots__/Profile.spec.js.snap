// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot renders 1`] = `
<div>
  <div
    className="media"
  >
    <img
      className="mr-3"
      height={64}
      src="blabl blah"
      width={64}
    />
    <div
      className="media-body"
    >
      <h5 />
      <p />
      <h5>
        Tweets
      </h5>
      <div>
        <div
          className="spinner-border text-primary"
        />
      </div>
    </div>
  </div>
</div>
`;
